feat: DXDP-5257 | Adding UL theme Page Layout Custom Component

grandmaester opened the pull request:
### Description

This PR introduces a Page Layout Component(ULThemePageLayout) designed custom using Auth0 Universal theme tokens and considering tenant branding settings.

<img width="1721" height="955" alt="image" src="https://github.com/user-attachments/assets/520c6ac5-a714-4b04-bb84-bd6e983b39bb" />

<img width="1726" height="953" alt="image" src="https://github.com/user-attachments/assets/99403ab4-b5b1-4900-b9c5-f69c0276ba7a" />


**Files Modified:**

1. /src/components/ULThemePageLayout.tsx - New UL Page Layout component with Auth0 theme 
2. /src/utils/theme/themeFlatteners.ts - Universal process font size and font weight logic
3. /src/index.css - Theme variables updates and custom variant definition
4. /src/screens/login-id/index.tsx - Migrated to include new page layout component

**Auth0 Theme Variables:**

1.   --ul-theme-page-bg-page-layout - Page Alignment token
2.   --ul-theme-page-bg-background-color - Page Background Color token
3.   --ul-theme-page-bg-background-image-url - Page Background Image token

### Testing

**Unit Test Coverage:**

**Test Suite:** /src/components/__tests__/ULThemePageLayout.test.tsx
**Snapshot Tests: , Functional Tests:**

<img width="521" height="312" alt="image" src="https://github.com/user-attachments/assets/3be0796c-b6e1-4159-8536-d90e45a9f003" />

**ULThemePageLayout Tests Only:**

```npm test -- ULThemePageLayout ```

amberkamboj77 commented on src/components/ULThemePageLayout.tsx:
```diff
@@ -0,0 +1,30 @@
+  ...rest
+}: ULThemePageLayoutProps) => {
+  const themedStyles =
+    "flex px-(--ul-page-padding-inline) py-(--ul-page-padding-block) justify-page-layout bg-background-page bg-(image:--ul-theme-page-bg-background-image-url)";
```
Do we require `--ul-page-padding-inline` , `--ul-page-padding-block` css variables? Will tailwind utilities like `px-5, py-5` help here if its not dynamically changing?

grandmaester replied:
@amberkamboj77 This is an area of discussion. Currently, I am not sure if this will need dynamic values ever, but for the sake of keeping component dynamic for future, I am driving them with variables. let me know if this is overhead.

amberkamboj77 reviewed (commented):
Added some comments. Also can we test it while configuring advanced mode and changing these variables

grandmaester commented on src/screens/login-id/index.tsx:
```diff
@@ -17,14 +18,14 @@ const LoginIdScreen: React.FC = () => {
 
   return (
-    <div className="min-h-screen flex items-center justify-center px-10 py-20">
+    <ULThemePageLayout className="theme-universal">
```
Updated @amberkamboj77

grandmaester commented:
> Overall good work. Added some comments. Can we test it while configuring advanced mode and changing these variables?

Fixed the review comments, Added Unit Tests as well. Will Publish Advanced mode test results here!

amberkamboj77 reviewed (approved):
approved
